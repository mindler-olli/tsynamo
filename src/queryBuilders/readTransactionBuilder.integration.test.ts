import { DynamoDBDocumentClient } from "@aws-sdk/lib-dynamodb";
import { DDB } from "../../test/testFixture";
import { getDDBClientFor, startDDBTestContainer } from "../../test/testUtil";
import { Tsynamo } from "../index";

describe("ReadTransactionBuilder", () => {
  let tsynamoClient: Tsynamo<DDB>;
  let ddbClient: DynamoDBDocumentClient;

  beforeAll(async () => {
    const testContainer = await startDDBTestContainer();
    ddbClient = await getDDBClientFor(testContainer);

    tsynamoClient = new Tsynamo<DDB>({
      ddbClient,
    });
  });

  it("handles transaction with gets", async () => {
    const trx = tsynamoClient.createReadTransaction();

    trx.addItem({
      Get: tsynamoClient.getItem("myTable").keys({
        userId: "123",
        dataTimestamp: 222,
      }),
    });

    trx.addItem({
      Get: tsynamoClient
        .getItem("myOtherTable")
        .keys({
          userId: "123",
          stringTimestamp: "111",
        })
        .attributes(["userId"]),
    });

    trx.addItem({
      Get: tsynamoClient.getItem("myTable").keys({
        userId: "1111111",
        dataTimestamp: 2222,
      }),
    });

    const result = await trx.execute();
    expect(result).toMatchSnapshot();
  });
});
