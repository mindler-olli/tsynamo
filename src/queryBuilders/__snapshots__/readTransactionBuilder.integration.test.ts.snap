// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReadTransactionBuilder > handles transaction with gets 1`] = `
[
  [
    "myTable",
    {
      "dataTimestamp": 222,
      "someBoolean": true,
      "somethingElse": 2,
      "userId": "123",
    },
  ],
  [
    "myOtherTable",
    {
      "userId": "123",
    },
  ],
  [
    "myTable",
    undefined,
  ],
]
`;
